### ISSUE TITLE
runtime: crashes on Android O

### ISSUE THREAD

**Issue by zx2c4** [golang/go#23750]
Android O has a seccomp filter.

Unfortunately it disallows use of `open` and `epoll_wait`, which are used by the Go runtime. Fortunately, it allows for `openat` and `epoll_pwait`, which map directly to the former ones by passing them the right arguments.

So far as I can see, these support `open` and `epoll_wait`:

- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/arm_app_policy.cpp
- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/mips_app_policy.cpp
- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/x86_app_policy.cpp

So far as I can see, these support `openat` and `epoll_pwait`:

- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/arm64_app_policy.cpp
- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/mips64_app_policy.cpp
- https://android.googlesource.com/platform/bionic/+/master/libc/seccomp/x86_64_app_policy.cpp

In other words, 64-bit architectures are affected. ARM64 was already fixed. This CL fixes the remaining two architectures: https://go-review.googlesource.com/c/go/+/92895

**Comment by ianlancetaylor** [comment]
Marking for 1.9.5, but note that this is also not fixed in 1.10.  We need some way to get a sense of whether the patches will work on a range of systems; likely they will but some experience is needed.  Probably best to commit them after 1.10 branches, and then consider backports to 1.10.1 and 1.9.5.

**Comment by zx2c4** [comment]
@ianlancetaylor This should probably be reopened until https://github.com/golang/go/commit/04e6ae6bc3ac9739568b0f1225ee5e2d53cba919 is backported to 1.9 and 1.10 branches.

**Comment by andybons** [comment]
CL 92895 OK for Go 1.10.1

**Comment by ianlancetaylor** [comment]
Looks like we should also include CL 100877 and 100878 in 1.10.1.

**Comment by andybons** [comment]
Done.
